### ISSUE TITLE
Building Android with EAS Build, the build fails due to Gradle with Bugsnag

### ISSUE THREAD

**Issue by amalasyrafs** [bugsnag/bugsnag-expo#110]
### Describe the bug
When  I try to run EAS build on Android, I'm getting this error message.

`Running 'gradlew :app:bundleRelease' in /home/expo/workingdir/build/android
Downloading https://services.gradle.org/distributions/gradle-7.5.1-all.zip
10%
20%.
30
%.
40%
50
%.
60%.
70%.
80%.
90
%.
100%
Welcome to Gradle 7.5.1!
Here are the highlights of this release:
 - Support for Java 18
 - Support for building with Groovy 4
 - Much more responsive continuous builds
 - Improved diagnostics for dependency resolution
For more details see https://docs.gradle.org/7.5.1/release-notes.html
To honour the JVM settings for this build a single-use Daemon process will be forked. See https://docs.gradle.org/7.5.1/userguide/gradle_daemon.html#sec:disabling_the_daemon.
Daemon will be stopped at the end of the build
> Task :react-native-gradle-plugin:pluginDescriptors
> Task :react-native-gradle-plugin:processResources
> Task :react-native-gradle-plugin:compileKotlin
w: /home/expo/workingdir/build/node_modules/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/ReactExtension.kt: (10, 37): 'BaseVariant' is deprecated. Deprecated in Java
w: /home/expo/workingdir/build/node_modules/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/ReactExtension.kt: (119, 30): 'BaseVariant' is deprecated. Deprecated in Java
w: /home/expo/workingdir/build/node_modules/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/ReactExtension.kt: (135, 26): 'BaseVariant' is deprecated. Deprecated in Java
w: /home/expo/workingdir/build/node_modules/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/ReactExtension.kt: (157, 32): 'BaseVariant' is deprecated. Deprecated in Java
w: /home/expo/workingdir/build/node_modules/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/ReactExtension.kt: (163, 31): 'BaseVariant' is deprecated. Deprecated in Java
w: /home/expo/workingdir/build/node_modules/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/ReactExtension.kt: (171, 36): 'BaseVariant' is deprecated. Deprecated in Java
w: /home/expo/workingdir/build/node_modules/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/ReactPlugin.kt: (114, 48): 'reactRoot: DirectoryProperty' is deprecated. reactRoot was confusing and has been replace with root to point to your root project and reactNativeDir to point to the folder of the react-native NPM package
w: /home/expo/workingdir/build/node_modules/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/TaskConfiguration.kt: (10, 37): 'ApplicationVariant' is deprecated. Deprecated in Java
w: /home/expo/workingdir/build/node_modules/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/TaskConfiguration.kt: (11, 37): 'BaseVariant' is deprecated. Deprecated in Java
w: /home/expo/workingdir/build/node_modules/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/TaskConfiguration.kt: (12, 37): 'LibraryVariant' is deprecated. Deprecated in Java
w: /home/expo/workingdir/build/node_modules/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/TaskConfiguration.kt: (27, 51): 'BaseVariant' is deprecated. Deprecated in Java
w: /home/expo/workingdir/build/node_modules/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/TaskConfiguration.kt: (130, 12): 'ApplicationVariant' is deprecated. Deprecated in Java
w: /home/expo/workingdir/build/node_modules/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/TaskConfiguration.kt: (131, 12): 'LibraryVariant' is deprecated. Deprecated in Java
w: /home/expo/workingdir/build/node_modules/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/TaskConfiguration.kt: (251, 14): 'BaseVariant' is deprecated. Deprecated in Java
> Task :react-native-gradle-plugin:compileJava NO-SOURCE
> Task :react-native-gradle-plugin:classes
> Task :react-native-gradle-plugin:inspectClassesForKotlinIC
> Task :react-native-gradle-plugin:jar
[stderr] 
FAILURE: Build failed with an exception.
[stderr] 
* Where:
[stderr] 
Build file '/home/expo/workingdir/build/android/app/build.gradle' line: 375
[stderr] 
* What went wrong:
[stderr] 
A problem occurred evaluating project ':app'.
[stderr] 
> Failed to apply plugin 'com.bugsnag.android.gradle'.
[stderr] 
   > Using com.bugsnag.android.gradle:8+ with Android Gradle Plugin < 8 is not supported. Either upgrade the Android Gradle Plugin to 8, or use an earlier version of the BugSnag Gradle Plugin. For more information about this change, see https://docs.bugsnag.com/build-integrations/gradle/
[stderr] 
* Try:
[stderr] 
> Run with --stacktrace option to get the stack trace.
[stderr] 
> Run with --info or --debug option to get more log output.
[stderr] 
> Run with --scan to get full insights.
[stderr] 
* Get more help at https://help.gradle.org
[stderr] 
BUILD FAILED
[stderr] 
in 1m 40s
5 actionable tasks: 5 executed
Error: Gradle build failed with unknown error. See logs for the "Run gradlew" phase for more information.`

I tried removing bugsnag package from my project, and the build works.
I have also tried downgrading bugsnag package to version 44 ( this is my last project version before bumping up Expo SDK) and the build works.

It seems like expo build is on SDK 47 is using Gradle 7.5.1.

My project have these dependencies from bugsnag:
"@bugsnag/expo": "^47.0.0",
"@bugsnag/plugin-expo-eas-sourcemaps": "^47.0.0",
"@bugsnag/source-maps": "^2.3.1",

### Steps to reproduce
Im not sure to report to expo or bugsnag but since the error seems like caused by bugsnag specifically, I decided to report here instead.

### Environment
* Bugsnag version: 47.0.0 & 47.1.0 (tried both)
* Expo SDK version (e.g. SDK 44): SDK47
* OS:  Android

**Comment by amalasyrafs** [comment]
I guess worth also mentioning that when I build on Android with the same package version on Dec 2022, the build works.

Im not sure why bugsnag expect gradle version 8.

**Comment by amalasyrafs** [comment]
Update: the build works when remove @bugsnag/plugin-expo-eas-sourcemaps package. My guess is that it expect Gradle 8 as per the error.

**Comment by thorbenprimke** [comment]
@amalasyrafs - I just ran into the same issue today. [It's due to a new release of the gradle plugin](https://github.com/bugsnag/bugsnag-android-gradle-plugin/releases). 

At least for local AS builds (using expo prebuild), I was able to tweak it

```
        dependencies {
            classpath 'com.bugsnag:bugsnag-android-gradle-plugin:[7.4.1]'
        }

```

instead of

```
        dependencies {
            classpath 'com.bugsnag:bugsnag-android-gradle-plugin:[7.3.0,)'
        }
```

**Comment by thorbenprimke** [comment]
The solution here it try to patch the config plugin locally so that EAS builds work as well.

**Comment by thorbenprimke** [comment]
Yea, I fixed the bugsnag gradle plug-in version to the [latest before Gradle 8](https://github.com/bugsnag/bugsnag-android-gradle-plugin/releases). Below the patch package. My build on EAS was successful. I feel like the config plug-in should be updated to protect against this breaking change. At least make it a configurable param to opt-in to use gradle 8. @lemnik what do you think?

```
diff --git a/node_modules/@bugsnag/plugin-expo-eas-sourcemaps/src/android.js b/node_modules/@bugsnag/plugin-expo-eas-sourcemaps/src/android.js
index 5f6279c..b23a06a 100644
--- a/node_modules/@bugsnag/plugin-expo-eas-sourcemaps/src/android.js
+++ b/node_modules/@bugsnag/plugin-expo-eas-sourcemaps/src/android.js
@@ -52,7 +52,7 @@ function injectDependencies (script) {
             mavenCentral()
         }
         dependencies {
-            classpath 'com.bugsnag:bugsnag-android-gradle-plugin:[7.3.0,)'
+            classpath 'com.bugsnag:bugsnag-android-gradle-plugin:[7.4.1]'
         }
     }
 
 ```

**Comment by thorbenprimke** [comment]
Nevermind - already fixed - https://github.com/bugsnag/bugsnag-expo/releases - strange that it didn't pull this in automatically since it's a patch version change.

**Comment by amalasyrafs** [comment]
Yeah it was fixed with the patch and I manage to create a build on Android again.

But your patch also helped a lot.
Thanks @thorbenprimke . I was able to learn something new today.

**Comment by luke-belton** [comment]
Hi all - thanks for raising this with us, this was also discussed over on https://github.com/bugsnag/bugsnag-expo/issues/102. As mentioned in the thread above, this was caused by a [beta release](https://github.com/bugsnag/bugsnag-android-gradle-plugin/releases/tag/v8.0.0-beta01) of our support for Android Gradle Plugin `v8`.

The fix has been released for all versions of Expo we currently maintain support for and ensures that this won't happen again for future `v8` releases of the BugSnag Android Gradle Plugin. The issue also doesn't impact earlier versions of `bugsnag-expo` (i.e. `v44` and below).
* https://github.com/bugsnag/bugsnag-expo/releases/tag/v47.1.1
* https://github.com/bugsnag/bugsnag-expo/releases/tag/v46.0.3
* https://github.com/bugsnag/bugsnag-expo/releases/tag/v45.1.3

**Comment by lucasklaassen** [comment]
@luke-belton can you please publish 47.1.1 to npm?

**Comment by luke-belton** [comment]
Hi @lucasklaassen - thanks for raising this with us. We're working on getting all of the above releases published on npm (currently only `v45.1.3` is available). We'll update this thread when they're available.

**Comment by imjoehaines** [comment]
@lucasklaassen This only required an update to `@bugsnag/plugin-expo-eas-sourcemaps` where v47.1.1 is already on npm

`@bugsnag/expo` itself did not change, so there's no corresponding version on npm

So to update you only need to bump the version of `@bugsnag/plugin-expo-eas-sourcemaps` to v47.1.1
